nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/crypto/ScryptCipherProvider.java

```diff
@@ -78,6 +79,17 @@ public ScryptCipherProvider(int n, int r, int p) {
         if (p < DEFAULT_P) {
             logger.warn("The provided parallelization factor {} is below the recommended minimum {}", p, DEFAULT_P);
         }
+
+        if (!isPValid(r, p)) {
+            logger.warn("Based on the provided block size {}, the provided parallelization factor {} is out of bounds", r, p);
+            throw new IllegalArgumentException("Invalid p value exceeds p boundary");
+        }
+    }
+
+    public static boolean isPValid(int r, int p) {
+        // Calculate p boundary
+        double pBoundary = ((Math.pow(2, 32)) - 1) * (32.0 / (r * 128));
```

Comment:
I think we may need to validate that `r` is a positive integer as well, because submitting a negative number or 0 here will cause problems.